tests/conftest.py

```diff
@@ -93,6 +93,37 @@ def generate_csv_no_header(tmpdir_factory):
     return str(fn)
 
 
+@pytest.fixture(scope="session")
+def generate_csv_float(tmpdir_factory):
```

Comment:
This is my own opinion, but you should use the same method you used for `generate_csv_float_no_header` here, instead of the `csv` import.

Reply:
Since `def generate_csv` also used csv as import, I would change both of them in the same time which probably should not be under this PR. Also, I'm not sure how to generate a csv which has header which out use `csv`. Should I just make first line different?

Reply:
Yes, you can do the same as your other example -- there's even a comment note on how to generate a header; it's just the first line.